woocommerce/woocommerce-android
# Issue/7876 generate variations tracks

Closes: #7876

### Why
We want to measure the usage of the generate all variants feature and to understand the importance of supporting generating more than 100 variations.

### Description
This PR adds two track events. The first one, `product_variation_generation_requested`,  is to help us understand the adoption of the generate all variants feature. The second one, `product_variation_generation_limit_reached`, is to help us understand how important it is to increase the current limit for creating variations in bulk.
1. `PRODUCT_VARIATION_GENERATION_REQUESTED` when a user requests to generate all variations
2. `PRODUCT_VARIATION_GENERATION_LIMIT_REACHED` when a user requested to generate more than 100 variations.
3.  `PRODUCT_VARIATION_GENERATION_CONFIRMED` when the user confirms the generation.
4. `PRODUCT_VARIATION_GENERATION_SUCCESS`  when generated variations are successfully created.
5. `PRODUCT_VARIATION_GENERATION_FAILURE` when generated variations are not created.

### Testing instructions

TC1
1. Create a new product variation
2. Add some variations attributes considering that the cartesian product of the attributes should be <= `100`
3. Tap on Generate All Variations
4.  Check that the `product_variation_generation_requested` event is sent
5. On the confirmation dialog, tap on OK
6. Check that the `product_variation_generation_confirmed` event is sent
7. Check that the `product_variation_generation_success` event is sent

TC2
1. Create a new product variation
2. Add some variations attributes considering that the cartesian product of the attributes should be <= `100`
3. Tap on Generate All Variations
4.  Check that the `product_variation_generation_requested` event is sent
5. Turn off wifi on the device
6. On the confirmation dialog, tap on OK
7. Check that the `product_variation_generation_confirmed` event is sent
8. Check that the `product_variation_generation_failure` event is sent

TC3
1. Create a new product variation
2. Add some variations attributes considering that the cartesian product of the attributes should be > `100`
3. Tap on Generate All Variations
4. Check that the `product_variation_generation_limit_reached` event is sent

### Images/gif
```
🔵 Tracked: product_variation_generation_requested, Properties: {"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```
```
 🔵 Tracked: product_variation_generation_confirmed, Properties: {"variations_count":3,"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```
```
🔵 Tracked: product_variation_generation_success, Properties: {"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```
```
 🔵 Tracked: product_variation_generation_limit_reached, Properties: {"variations_count":240,"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```
```
 🔵Tracked: product_variation_generation_failure, Properties: {"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```
